[Bug][Packaging][requirements.txt omits both opentelemetry deps, so a clean install cannot import swarms and the Python package CI job is red]

Any contributor following the standard clone + `pip install -r requirements.txt` flow gets an environment where `import swarms` raises `ModuleNotFoundError`. 62 of 125 test modules under `tests/` then fail at collection, and the `Python package` workflow — which installs via `requirements.txt` — has been failing on master for this reason.

Published PyPI wheels are **not** affected. `poetry build` reads `pyproject.toml`, and `swarms==14.0.0` correctly declares both OpenTelemetry packages in `requires_dist`. This is requirements.txt / pyproject drift affecting the contributor workflow and CI, not an end-user install break.

## Root cause

`pyproject.toml:76-77` declares the dependencies:

```toml
opentelemetry-sdk = "*"
opentelemetry-exporter-otlp-proto-http = "*"
```

`requirements.txt` — the complete 15-line file — lists neither:

```
asyncio>=3.4.3,<5.0
toml
loguru
pydantic==2.12.5
tenacity
rich
psutil
python-dotenv
PyYAML
networkx
httpx
requests
litellm==1.76.1
mcp>=1.28.1,<2.0.0
schedule
```

`swarms/telemetry/otel.py:16-23` imports the SDK and the OTLP HTTP exporter unguarded at module scope, and that module sits on the eager path of the top-level package: `swarms/__init__.py:5` → `swarms.telemetry.bootup` → `swarms/telemetry/__init__.py:6` (`from swarms.telemetry.otel import log_agent_data`). There is no `try/except ImportError` and no optional-extra gating.

## Reproducer

```bash
python3 -m venv /tmp/clean
/tmp/clean/bin/pip install -r requirements.txt
/tmp/clean/bin/python -c "import swarms"
```

Output on `main` @ `297ae6b7` (v14.0.0), Python 3.12.3:

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File ".../swarms/__init__.py", line 5, in <module>
    from swarms.telemetry.bootup import bootup  # noqa: E402, F403
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File ".../swarms/telemetry/__init__.py", line 6, in <module>
    from swarms.telemetry.otel import log_agent_data
  File ".../swarms/telemetry/otel.py", line 16, in <module>
    from opentelemetry import context, trace
ModuleNotFoundError: No module named 'opentelemetry'
```

Downstream, in the same environment:

```
$ /tmp/clean/bin/python -m pytest tests/ --collect-only -q | tail -3
!!!!!!!!!!!!!!!!!!! Interrupted: 62 errors during collection !!!!!!!!!!!!!!!!!!!
63 tests collected, 62 errors in 5.36s
```

Installing exactly the two missing packages resolves it, confirming nothing else is involved.

## Already visible in CI

The `Python package` workflow run on master (v14) fails at `Test with pytest`:

```
collected 167 items / 106 errors
E   ModuleNotFoundError: No module named 'opentelemetry'   (tests/telemetry/*, 4x)
E   ModuleNotFoundError: No module named 'swarms'          (102x)
```

The 102 `No module named 'swarms'` errors are a second, independent defect in the same workflow: `.github/workflows/python-package.yml:33` installs only the dependency list and never installs the project itself, so pytest's per-directory `sys.path` insertion cannot resolve `swarms` for test files outside the repo root.

No workflow performs a clean-install import smoke test — `grep -rn "import swarms" .github/workflows/` returns nothing, which is why this class of drift fails silently.

## Suggested change

1. Add the two missing dependencies to `requirements.txt`:

```
opentelemetry-sdk
opentelemetry-exporter-otlp-proto-http
```

`setuptools` (declared at `pyproject.toml:60`) is also absent, and `pydantic` is pinned to `==2.12.5` in requirements.txt while left unconstrained in pyproject — worth reconciling in the same pass, or generating `requirements.txt` from `poetry export` so the drift stops recurring.

2. In `.github/workflows/python-package.yml:33`, install the project as well so the tests can import it:

```yaml
if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
pip install -e .
```

3. Add an import smoke test so this fails loudly next time:

```yaml
- name: Import smoke test
  run: python -c "import swarms; print(swarms.__name__)"
```

## Environment

swarms `main` @ `297ae6b7` (v14.0.0), Python 3.12.3, Linux.

--- a/requirements.txt
+++ b/requirements.txt
@@ -13,3 +13,5 @@ requests
 litellm==1.76.1
 mcp>=1.28.1,<2.0.0
 schedule
+opentelemetry-sdk
+opentelemetry-exporter-otlp-proto-http

